services/slam/fake/slam_test.go

```diff
@@ -170,3 +240,88 @@ func reverse[T any](slice []T) []T {
+	getPositionResults := []spatialmath.Pose{}
+	getInternalStateResults := []int{}
+
+	// Call GetPointCloudMapStream twice for every testData artifact
+	for i := 0; i < testDataCount*2; i++ {
+		pInFrame, err := slamSvc.Position(context.Background(), slamSvc.Name, extra)
+		test.That(t, err, test.ShouldBeNil)
+		positionResults = append(positionResults, pInFrame.Pose())
+
+		p, _, err := slamSvc.GetPosition(context.Background(), slamSvc.Name)
+		test.That(t, err, test.ShouldBeNil)
+		getPositionResults = append(getPositionResults, p)
+
+		data, err := slamSvc.GetInternalState(context.Background(), slamSvc.Name)
+		test.That(t, err, test.ShouldBeNil)
+		getInternalStateResults = append(getInternalStateResults, len(data))
+
+		f, err := slamSvc.GetPointCloudMapStream(context.Background(), slamSvc.Name)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, f, test.ShouldNotBeNil)
+		pcd, err := helperConcatenateChunksToFull(f)
+		test.That(t, err, test.ShouldBeNil)
+		pc, err := pointcloud.ReadPCD(bytes.NewReader(pcd))
+		test.That(t, err, test.ShouldBeNil)
+		vObj, err := vision.NewObject(pc)
+		test.That(t, err, test.ShouldBeNil)
+
+		getMapPcdResults = append(getMapPcdResults, vObj.MetaData().MaxX)
+		test.That(t, err, test.ShouldBeNil)
+	}
+
+	getPositionResultsFirst := getPositionResults[len(getPositionResults)/2:]
+	getPositionResultsLast := getPositionResults[:len(getPositionResults)/2]
+
+	positionResultsFirst := positionResults[len(positionResults)/2:]
+	positionResultsLast := positionResults[:len(positionResults)/2]
+
+	getInternalStateResultsFirst := getInternalStateResults[len(getInternalStateResults)/2:]
+	getInternalStateResultsLast := getInternalStateResults[:len(getInternalStateResults)/2]
+
+	// Confirm that the first half of the
+	// results equal the last.
+	// This proves that each call to GetPointCloudMapStream
+	// advances the test data (both for GetPointCloudMapStream & other endpoints)
+	// over a dataset of size maxDataCount that loops around.
+	test.That(t, positionResultsFirst, test.ShouldResemble, positionResultsLast)
+	test.That(t, getPositionResultsFirst, test.ShouldResemble, getPositionResultsLast)
+	test.That(t, getInternalStateResultsFirst, test.ShouldResemble, getInternalStateResultsLast)
+
+	// Confirm that the first half of the
+	// results does NOT equal the last half in reverse.
+	// This proves that each call to GetPointCloudMapStream
+	// advances the test data (both for GetPointCloudMapStream & other endpoints)
+	// over a dataset of size maxDataCount that loops around.
+	test.That(t, positionResultsFirst, test.ShouldNotResemble, reverse(positionResultsLast))
+	test.That(t, getPositionResultsFirst, test.ShouldNotResemble, reverse(getPositionResultsLast))
+	test.That(t, getInternalStateResultsFirst, test.ShouldNotResemble, reverse(getInternalStateResultsLast))
+
+	supportedMimeTypes := []string{rdkutils.MimeTypePCD, rdkutils.MimeTypeJPEG}
+	test.That(t, supportedMimeTypes, test.ShouldContain, mimeType)
```

Comment:
This is only tested in one place with MimeType PCD and is not necessary for stream tests